### DISCUSSION TITLE
Simple query

### DISCUSSION THREAD

**Discussion by JoeGills** [data-interaction-fed24/server#8] [Q&A] [answered]
// Get data by id... 

app.get("/api/my_table/:id", async (req, res) => {
  const id = req.params.id;

  try {
    const queryText = "SELECT * FROM my_table WHERE id = $1";
    const values = [id];
    const result = await pool.query(queryText, values);

**Why did we use   const id = req.params.id; outside try{} ?**

Instead can we use like this:

app.get("/api/my_table/:id", async (req, res) => {
  try {
    const id = req.params.id;
    const queryText = "SELECT * FROM my_table WHERE id = $1";
    const { rows } = await pool.query(queryText, [id]);

**Comment by Liam-hi** [comment, marked as answer]
Good question!

So the try-block is the block of code which exceptions can occur. 
In our case, we are not expecting that `const id = req.params.id;` will raise any errors and it's unlikely that `const queryText = "SELECT * FROM my_table WHERE id = $1";` and `const values = [id];` will raise any errors but it's more or less useful to have it inside the try-block for debugging purposes.

**Comment by JoeGills** [comment]
Thank you for the answer
